pygeoapi/linked_data.py

```diff
@@ -171,35 +171,89 @@ def geojson2geojsonld(config, data, dataset, identifier=None):
     :param data: dict of data:
     :param dataset: dataset identifier
     :param identifier: item identifier (optional)
+    :param identifier_field: item identifier_field (optional)
 
     :returns: string of rendered JSON (GeoJSON-LD)
     """
     context = config['resources'][dataset].get('context', [])
-    data['id'] = (
-        '{}/collections/{}/items/{}' if identifier
-        else '{}/collections/{}/items'
-    ).format(
-        *[config['server']['url'], dataset, identifier]
-    )
+    geojsonld = config['resources'][dataset].get('geojsonld', True)
+
+    if identifier:
+    # Single geojsonld
+        if not geojsonld:
+            data, geocontext = make_jsonld(data)
+            data[identifier_field] = identifier
+        else:
+            data['id'] = identifier
+    else:
+    # Collection of geojsonld
+        data['@id'] = '{}/collections/{}/items/'.format(
+            config['server']['url'], dataset)
+        for i, feature in enumerate(data['features']):
+            identifier = feature.get(identifier_field, feature.get('properties').get(identifier_field, ''))
+            if not is_url(str(identifier)):
+                identifier = '{}/collections/{}/items/{}'.format(
+                            config['server']['url'], dataset, feature['id'])
+
+            if not geojsonld:
+                feature, geocontext = make_jsonld(feature)
+                # Note: @id or https://schema.org/url or both or something else?
+                feature[identifier_field] = identifier
+            else:
+                feature['id'] = identifier
+           
+            data['features'][i] = feature
```

Comment:
Not sure if this is necessary, or if `feature` is already being mutated in the enumeration?

Reply:
I think it happens on line 199 when the `feature` is re-declared from the custom json-ld context. This felt like the simplest way to ensure the feature retained was the updated one.